core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -223,6 +224,20 @@ public void runClusterServices(@Nonnull Stack stack, @Nonnull Cluster cluster, L
         }
     }
 
+    public void updateClusterConfigs(@Nonnull Stack stack, @Nonnull Cluster cluster, List<String> candidateAddresses) {
+        try {
+            Set<Node> nodes = stackUtil.collectReachableNodes(stack);
+            GatewayConfig primaryGatewayConfig = gatewayConfigService.getPrimaryGatewayConfig(stack);
+            List<GatewayConfig> gatewayConfigs = gatewayConfigService.getAllGatewayConfigs(stack);
+            SaltConfig saltConfig = createSaltConfig(stack, cluster, primaryGatewayConfig, gatewayConfigs, nodes);
+            ExitCriteriaModel exitCriteriaModel = nonCancellableModel();
```

Comment:
I think this should be `ClusterDeletionBasedExitCriteriaModel`

Reply:
Ok didn't quite understand the distinction

Reply:
fixed